### PR TITLE
Issue 9381 fix navigation on the error modal  for incomplete application step

### LINKED ISSUE #9381: Review & Exit shows the wrong modal
Non-NS: 
I partially fill in a page and click _review_:
image

I partially fill in a page and click _previous_:
image

NS: 
I partially fill in a page and click _review and exit_:
image

I partially fill in a page and click _back_:
image

In NS, clicking _review and exit_ should give me the same modal as clicking _review_ does in non-NS.

### PR DESCRIPTION
**Pull request by haltmar**
### Description

Conditionally set navigation and display text on the error modal dialog to either previous block or review page based on whether user click on "back" or "review and exit" button the the page.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [ ] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`

#### Database evolutions

Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#writing-database-evolutions)

N/A

#### Durable jobs

Read the docs [here](https://github.com/civiform/civiform/wiki/Database#durable-jobs-for-data-updates)

N/A

#### User visible changes

- [ ] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [ ] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [ ] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order
- [ ] Manually tested with a screen reader if the feature is applicant-facing. See [screen reader testing](https://github.com/civiform/civiform/wiki/Testing#screen-reader-testing)

#### New Features

N/A

### Instructions for manual testing

create a program with multiple steps, one of which contains required questions. Alternatively,  you can import the attached test program json.

start application, go the step which contains required questions, 
click on Back,  you should see the dialog saying

Click on "Review and exit", you should see the dialog saying " ", click on "go to review", UI should take you to the review page (whereas without the fix, user can only go back to previous page)

### Issue(s) this completes

Fixes #9381

### CODE REVIEW

**Comment by haltmar** [comment]
[test-program-for-navigation-flows-exported.json](https://github.com/user-attachments/files/18203330/test-program-for-navigation-flows-exported.json)

test Program json file exported from existing test case for reviewer's convenience.

File: browser-test/src/support/applicant_questions.ts [hunk 1]
```diff
@@ -960,16 +960,38 @@ export class ApplicantQuestions {
       'This question is required',
     )
   }
+  async expectErrorOnReviewModal(northStarEnabled = false) {
+    if (northStarEnabled) {
```
Comment (dkatzz) [inline browser-test/src/support/applicant_questions.ts:964]: **suggestion:** Thank you for adding all of these assertions, but I think we can assert the same text as non-northstar (i.e. don't have to add text assertions for everything). You could also consider creating variables for the text, like `modalTitle`, `modalContent`, and `modalButton` and then using that in the branches.

File: server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java [hunk 2]
```diff
@@ -112,10 +113,36 @@ public String render(Request request, ApplicationBaseViewParams applicationParam
       context.setVariable(
```
Comment (dkatzz) [inline server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java:113]: **suggestion:** This is a bit hard to parse. I think one way to make this a little more clear would be to create two functions `setContextForReview` or `setContextForPrevious` and then you can branch which function gets called based on the applicationParams.errorDisplayMode(). And beyond that you could even make a helper function `setContextForFormModal`, which gets called from each of those sub-functions and would take in variables, like errorModalFormAction, errorModalTitle, errorModalContent, errorModalButtonText and errorModalDataRedirectTo and set the context for each. And then each function would set their particular form action then call the helper function (if you think its cleaner to create that)

File: browser-test/src/support/applicant_questions.ts (hunk 1 above)
Comment (haltmar) [reply browser-test/src/support/applicant_questions.ts:971]: Thanks for the suggestion.  refactored the method. PTAL.

File: server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java (hunk 2 above)
Comment (haltmar) [reply server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java:114]: thanks for the suggestion.  refactored the code. PTAL
